packages/iris-grid/src/IrisGridRenderer.ts

```diff
@@ -151,25 +158,93 @@ class IrisGridRenderer extends GridRenderer {
-    const { modelColumns, modelRows } = metrics;
-    const modelRow = getOrThrow(modelRows, row);
+    const { metrics } = state;
+    const { modelColumns } = metrics;
     const modelColumn = modelColumns.get(column);
     if (modelColumn === undefined) {
       return;
     }
-    const value = model.valueForCell(modelColumn, modelRow);
-    if (TableUtils.isTextType(model.columns[modelColumn]?.type)) {
-      if (value === null || value === '') {
-        const originalFont = context.font;
-        context.font = `italic ${originalFont}`;
-        const displayValue = value === null ? 'null' : 'empty';
-        super.drawCellContent(context, state, column, row, displayValue);
-        context.font = originalFont;
-        return;
-      }
+
+    const cellRenderer = this.getCellRenderer('text');
+    cellRenderer.drawCellContent(context, state, column, row);
+  }
+
+  getCellOverflowButtonPosition({
+    mouseX,
+    mouseY,
+    metrics,
+    theme,
+  }: {
+    mouseX: Coordinate | null;
+    mouseY: Coordinate | null;
+    metrics: GridMetrics | undefined;
+    theme: GridThemeType;
+  }): {
+    left: Coordinate | null;
+    top: Coordinate | null;
+    width: number | null;
+    height: number | null;
+  } {
+    return this.textCellRenderer.getCellOverflowButtonPosition(
+      mouseX,
+      mouseY,
+      metrics,
+      theme
+    );
+  }
+
+  shouldRenderOverflowButton(state: IrisGridRenderState): boolean {
+    return this.textCellRenderer.shouldRenderOverflowButton(state);
+  }
+
+  drawCellOverflowButton(state: IrisGridRenderState): void {
+    const { context, mouseX, mouseY, theme } = state;
+    if (mouseX == null || mouseY == null) return;
+
+    if (!this.shouldRenderOverflowButton(state)) {
+      return;
+    }
+
+    const {
+      left: buttonLeft,
+      top: buttonTop,
+      width: buttonWidth,
+      height: buttonHeight,
+    } = this.getCellOverflowButtonPosition(state);
+
+    const {
+      cellHorizontalPadding,
+      overflowButtonColor,
+      overflowButtonHoverColor,
+    } = theme;
+
+    context.save();
+    if (
+      overflowButtonHoverColor != null &&
+      buttonLeft != null &&
+      buttonWidth != null &&
+      buttonTop != null &&
+      buttonHeight != null &&
+      mouseX >= buttonLeft &&
+      mouseX <= buttonLeft + buttonWidth &&
+      mouseY >= buttonTop &&
+      mouseY <= buttonTop + buttonHeight
+    ) {
+      context.fillStyle = overflowButtonHoverColor;
+    } else if (overflowButtonColor != null) {
+      context.fillStyle = overflowButtonColor;
+    }
+    const icon = this.getIcon(ICON_NAMES.CELL_OVERFLOW);
+    if (buttonLeft != null && buttonTop != null) {
+      context.translate(buttonLeft + cellHorizontalPadding, buttonTop + 2);
     }
-    super.drawCellContent(context, state, column, row);
+    context.fill(icon);
+
+    context.restore();
+  }
+
+  getCellRenderer(rendererType: CellRendererType): CellRenderer {
+    return this.textCellRenderer;
   }
```

Comment:
Shouldn't need to override this function, just use the parent which already returns `this.dataBarCellRenderer` or `this.textCellRenderer`